hubble/CFP-37472-flowlog-aggregation.md

```diff
@@ -0,0 +1,150 @@
+
+Providing an aggregation option as part of dynamic exporter would allow user to set the verbosity of logs they want to store, while also capturing the important signals from the cluster networking. Consider the current scenario where `client-1` and `client-2` are communicating with pod `server` over TCP. Say, both client pods are on same node. We expect to see entries like this on the log file stored on the node over 30 second period.
+
+
+| source_pod | source_port | destination_pod | destination_port | protocol | flags   |
+|------------|-------------|-----------------|------------------|----------|---------|
+| client-1   | 12345       | server          | 80               | TCP      | SYN     |
+| server     | 80          | client-1        | 12345            | TCP      | SYN-ACK |
+| client-1   | 12345       | server          | 80               | TCP      | ACK     |
+| client-1   | 12345       | server          | 80               | TCP      | PSH     |
+| server     | 80          | client-1        | 12345            | TCP      | ACK     |
+| server     | 80          | client-1        | 12345            | TCP      | PSH     |
+| client-1   | 12345       | server          | 80               | TCP      | ACK     |
+| client-1   | 12345       | server          | 80               | TCP      | FIN     |
+| server     | 80          | client-1        | 12345            | TCP      | FIN-ACK |
+| client-1   | 12345       | server          | 80               | TCP      | ACK     |
+| client-2   | 23456       | server          | 80               | TCP      | SYN     |
+| server     | 80          | client-2        | 23456            | TCP      | SYN-ACK |
+| client-2   | 23456       | server          | 80               | TCP      | ACK     |
+| client-2   | 23456       | server          | 80               | TCP      | PSH     |
+| server     | 80          | client-2        | 23456            | TCP      | ACK     |
+| server     | 80          | client-2        | 23456            | TCP      | PSH     |
+| client-2   | 23456       | server          | 80               | TCP      | ACK     |
+| client-2   | 23456       | server          | 80               | TCP      | FIN     |
+| server     | 80          | client-2        | 23456            | TCP      | FIN-ACK |
+| client-2   | 23456       | server          | 80               | TCP      | ACK     |
+
+However, we can gleam the same level of information if the logs were aggregated as follows:
+
+| source_pod | source_port | destination_pod | destination_port | protocol | ingress_flow_count | egress_flow_count |
+|------------|-------------|-----------------|------------------|----------|--------------------|-------------------|
+| client-1   | 12345       | server          | 80               | TCP      | 4                 | 6                 |
+| client-2   | 23456       | server          | 80               | TCP      | 4                 | 6                 |
```

Comment:
Does this assume that we take the traffic direction into account, or just ignore the reply flows?

Reply:
I am thinking we should take into account `is_reply` and provide the ingress and egress count of "flows" observed.  
`egress_flow_count` : `client-1` ---> `server`
`ingress_flow_count` : `client-1` <--- `server`